zcash_transparent/src/builder.rs

```diff
@@ -378,4 +399,54 @@ impl Bundle<Unauthorized> {
             authorization: Authorized,
         })
     }
+
+    pub fn apply_external_signatures(
+        self,
+        #[cfg(feature = "transparent-inputs")] signatures: Vec<secp256k1::ecdsa::Signature>,
```

Comment:
I do not like making individual arguments to methods conditionally compiled in a public library API, as that leads to all sorts of annoying breakage. (We do this in a few public APIs, but as much as possible only in crates that are effectively "app crates" and thus less susceptible to feature-flag-at-a-distance changes; `zcash_transparent` is not such a crate).

Instead this entire method should be behind the feature flag, and should just append signatures, with separate `prepare` and `finalize` methods like [`orchard::builder`](https://github.com/zcash/orchard/blob/fcb14defd75c0dd79512289c17a1ac46b5001d3a/src/builder.rs#L970-L1098) has.